lib/Controller/OCSController.php

```diff
@@ -45,32 +49,52 @@ class OCSController extends \OCP\AppFramework\OCSController {
 	/** @var IURLGenerator */
 	private $urlGenerator;
 
-	public function __construct($appName,
-								IRequest $request,
-								IRootFolder $rootFolder,
-								$userId,
-								DirectMapper $directMapper,
-								IURLGenerator $urlGenerator) {
+	/** @var TemplateManager */
+	private $manager;
+
+	/**
+	 * OCS controller
+	 *
+	 * @param string $appName
+	 * @param IRequest $request
+	 * @param IRootFolder $rootFolder
+	 * @param string $userId
+	 * @param DirectMapper $directMapper
+	 * @param IURLGenerator $urlGenerator
+	 * @param TemplateManager $manager
+	 */
+	public function __construct(string $appName,
+		IRequest $request,
+		IRootFolder $rootFolder,
+		string $userId,
+		DirectMapper $directMapper,
+		IURLGenerator $urlGenerator,
+		TemplateManager $manager) {
 		parent::__construct($appName, $request);
 
-		$this->rootFolder = $rootFolder;
-		$this->userId = $userId;
+		$this->rootFolder   = $rootFolder;
+		$this->userId       = $userId;
 		$this->directMapper = $directMapper;
 		$this->urlGenerator = $urlGenerator;
+		$this->manager      = $manager;
 	}
 
 	/**
 	 * @NoAdminRequired
 	 *
+	 * Init an editing session
+	 *
 	 * @param int $fileId
+	 * @return DataResponse
+	 * @throws OCSNotFoundException|OCSBadRequestException
 	 */
-	public function create($fileId) {
+	public function create(int $fileId): DataResponse {
```

Comment:
yeah we can't do that yet if we support NC12/13 still

Reply:
:cry: